src/stdlib_strings.f90

```diff
@@ -290,5 +300,80 @@ elemental function ends_with_string_string(string, substring) result(match)
 
     end function ends_with_string_string
 
+    !> Slices the region between the input 'first' and 'last' index (both inclusive)
+    !> of the input 'string' by taking strides of length 'stride'
+    !> Returns a new string_type object
+    elemental function slice_string(string, first, last, stride) result(sliced_string)
+        type(string_type), intent(in) :: string
+        integer, intent(in), optional :: first, last, stride
+        type(string_type) :: sliced_string
+
+        sliced_string = string_type(slice(char(string), first, last, stride))
+
+    end function slice_string
+
+    !> Slices the region between the input 'first' and 'last' index (both inclusive)
+    !> of the input 'string' by taking strides of length 'stride'
+    !> Returns a new string
+    pure function slice_char(string, first, last, stride) result(sliced_string)
+        character(len=*), intent(in) :: string
+        integer, intent(in), optional :: first, last, stride
+        integer :: first_index, last_index, stride_vector, strides_taken, length_string, i, j
+        character(len=:), allocatable :: sliced_string
+
+        length_string = len(string)
+        if (length_string > 0) then
+            first_index = 1
+            last_index = length_string
+            stride_vector = 1
+
+            if (present(stride)) then
+                if (stride /= 0) then
+                    if (stride < 0) then
+                        first_index = length_string
+                        last_index = 1
+                    end if
+                    stride_vector = stride
+                end if
+            else
+                if (present(first) .and. present(last)) then
+                    if (last < first) then
+                        stride_vector = -1
+                    end if
+                end if
+            end if
+
+            if (present(first)) then
+                first_index =  first
+            end if
+            if (present(last)) then
+                last_index = last
+            end if
+
+            strides_taken = floor( real(last_index - first_index) / real(stride_vector) )
```

Comment:
This should be safe to do with integer devision
```suggestion
            strides_taken = (last_index - first_index) / stride_vector
```

Reply:
if last_index = 27
first_index = 26
stride_vector =-2
I want the output -1
that's why I added floor

Reply:
Should give -1 with integer division right away, I don't think there is need for floating point arithmetic here.